lib/rest-adapter.js

```diff
@@ -403,6 +411,11 @@ RestAdapter.prototype._registerMethodRouteHandlers = function(router,
     // Express 4.x only supports delete
     verb = 'delete';
   }
+
+  if (fastRouter.canRegister(route, handler)) {
+    handler.methodName = sharedMethod.name;
+    fastRouter[verb](route, handler);
```

Comment:
I think adding extra properties on the `handler` method makes it more difficult to understand what's going on, because the other parts of the code are depending on the fact that `fn.methodName` is defined, which is not obvious.

That's why I proposed an API that is an extension of the original Express API:

```js
// express:
router[verb](route, handler)

// proposed extension:
router[verb](route, handler, options) 
```

OTOH, this is a minor detail. Please considered my comment, and if you still feel that `handler.methodName` is better, then let's do it your way.